### ISSUE TITLE
test.py

### ISSUE THREAD

**Issue by st021350** [st021350/some_#2]
import pandas as pd
import theano
import keras as kr

from numpy import genfromtxt
from keras.models import Sequential
from keras.layers import Dense, Activation
from keras.optimizers import SGD, Adadelta
from keras.layers import Dropout
import numpy as np
import math
from sklearn.preprocessing import normalize

data = genfromtxt('grid.csv', delimiter=',')
data = data[np.where(data[:,2]>0.19),:][0]
data = data[np.where(data[:,2]<0.62),:][0]

points = data[:, 0:3]
ex_ = data[:, 3]
ey_ = data[:, 4]
ez_ = data[:, 5]

ex_ = ex_.reshape((-1, 1))
ey_ = ey_.reshape((-1, 1))
ez_ = ez_.reshape((-1, 1))

m_ex_ = min(ex_)
m_ey_ = min(ey_)
m_ez_ = min(ez_)

ex_n = (ex_ - min(ex_)) / (max(ex_) - min(ex_))
ey_n = (ey_ - min(ey_)) / (max(ey_) - min(ey_))
ez_n = (ez_ - min(ez_)) / (max(ez_) - min(ez_))

k_ex = (max(ex_) - min(ex_))
k_ey = (max(ey_) - min(ey_))
k_ez = (max(ez_) - min(ez_))

for i in range(1, 250, 1):
    np.random.seed(i)

    sgd = SGD(lr=0.01, decay=1e-6, momentum=0.5, nesterov=True)

    b_size = 2
    batch = 16
    ada = Adadelta(lr=0.05, rho=0.95, epsilon=1e-08, decay=0.0)
    l_size = 16

    Ex_ = Sequential()
    Ex_.add(Dense(l_size, input_dim=3, activation='tanh'))
    Ex_.add(Dropout(0.5))
    Ex_.add(Dense(l_size, activation='tanh'))
    Ex_.add(Dropout(0.5))
    Ex_.add(Dense(l_size, activation='tanh'))
    Ex_.add(Dropout(0.5))
    Ex_.add(Dense(1, activation='tanh'))

    Ex_.compile(loss='mean_squared_error', optimizer=sgd, metrics=['accuracy'])
    Ex_.fit(points, ex_n, batch_size=batch, nb_epoch=10, verbose=2)

    Ey_ = Sequential()
    Ey_.add(Dense(l_size, input_dim=3, activation='tanh'))
    Ey_.add(Dropout(0.5))
    Ey_.add(Dense(l_size, activation='tanh'))
    Ey_.add(Dropout(0.5))
    Ey_.add(Dense(l_size, activation='tanh'))
    Ey_.add(Dropout(0.5))
    Ey_.add(Dense(1, activation='tanh'))

    Ey_.compile(loss='mean_squared_error', optimizer=sgd, metrics=['accuracy'])
    Ey_.fit(points, ey_n, batch_size=batch, nb_epoch=10, verbose=2)

    Ez_ = Sequential()
    Ez_.add(Dense(l_size, input_dim=3, activation='tanh'))
    Ez_.add(Dropout(0.5))
    Ez_.add(Dense(l_size, activation='tanh'))
    Ez_.add(Dropout(0.5))
    Ez_.add(Dense(l_size, activation='tanh'))
    Ez_.add(Dropout(0.5))
    Ez_.add(Dense(1, activation='tanh'))

    Ez_.compile(loss='mean_squared_error', optimizer=sgd, metrics=['accuracy'])
    Ez_.fit(points, ez_n, batch_size=batch, nb_epoch=10, verbose=2)

    PI = 3.14159265359
    CHARGE = 1.602e-19
    MASS = 9.1e-31

    B = 0.1
    energy = 0.1  # в Эв
    dt = 10e-9  # в сек
    start_time = 0  # в сек
    every_step = 1
    every_step_print = 1

    nu_rw = 50000
    rotate_direction = -1
    w_rw = nu_rw * 2 * PI
    w_z = 1.7e+06 * 2 * PI
    w_cycl = 2.79926e+09 * 2 * PI
    alpha = CHARGE / MASS

    w_minus = 0.5*(w_cycl - math.sqrt(w_cycl*w_cycl - 2*w_z*w_z))
    w_plus = 0.5*(w_cycl + math.sqrt(w_cycl*w_cycl - 2*w_z*w_z))

#w_cycl = w_plus + w_minus
#w_rw = w_z - 3 * w_minus

    end_time = 1e-3 #2 * PI / w_minus;

    phi0 = PI / 2
    k = 1

    begin_x = 0.01
    begin_y = 0.01
    begin_z = 0.20

    end_x = 0.01
    end_y = 0.01
    end_z = 0.6

    delta_x = end_x - begin_x
    delta_y = end_y - begin_y
    delta_z = end_z - begin_z

    x = begin_x
    y = begin_y
    z = begin_z

    vTotal = 1e06 * math.sqrt(2 * energy * 1.602 / 9.1)

# всем частицам пишем испульс
    n_ = vTotal / math.sqrt(delta_x * delta_x + delta_y * delta_y + delta_z * delta_z)

    vx = delta_x * n_
    vy = delta_y * n_
    vz = delta_z * n_

    vxs = list()
    vys = list()
    vzs = list()
    vxs.append(vx)
    vys.append(vy)
    vzs.append(vz)

    xs = list()
    ys = list()
    zs = list()

    dts = list()
    dts.append(0)

    xs.append(x)
    ys.append(y)
    zs.append(z)

    nSteps = int(end_time / dt)

    i = -1
    while i < nSteps:
        i = i+1
    # model fields
        ex = (Ex_.predict(np.array([[x, y, z]]))[0][0]*k_ex + m_ex_)[0]
        ey = (Ey_.predict(np.array([[x, y, z]]))[0][0]*k_ey + m_ey_)[0]
        ez = (Ez_.predict(np.array([[x, y, z]]))[0][0]*k_ez + m_ez_)[0]

 #   ex_ = (Ex_.predict(np.array([[x, y, z]]))[0][0]*k_ex + m_ex_)[0]
 #   ey_ = (Ey_.predict(np.array([[x, y, z]]))[0][0]*k_ey + m_ey_)[0]
  #  ez_ = (Ez_.predict(np.array([[x, y, z]]))[0][0]*k_ez + m_ez_)[0]

        t = dt * i
        if (z > 0.2 and z < 0.4):

            a = -(108.37 - 1029.32 * z + 2920.45 * z * z - 2597 * z * z * z)
        else:
            a = 0
        ex = ex + a * math.cos(w_rw * t + phi0)
        ey = ey - a * math.sin(w_rw * t + phi0)

        tmp_vx = (vx * (
            1 - 0.25 * dt * dt * B * B * alpha * alpha) + dt * alpha * ex + alpha * B * dt * vy + 0.5 * dt * dt * alpha * alpha * B * ey) / (
                 1 + 0.25 * dt * dt * B * B * alpha * alpha)
        tmp_vy = (vy * (
        1 - 0.25 * alpha * alpha * B * B * dt * dt) + dt * alpha * ey - alpha * B * dt * vx - 0.5 * dt * dt * alpha * alpha * B * ex) / (
                 1 + 0.25 * alpha * alpha * B * B * dt * dt)

        vx = tmp_vx * (1 - dt * k)
        vy = tmp_vy * (1 - dt * k)

        vz = vz + dt * (alpha * ez - vz * k - a * (x * math.cos(w_rw * t + phi0) - y * math.sin(w_rw * t + phi0)))

    #if ((i % every_step == 0) and (t > start_time)):
        dts.append(dt * i)

    #if ((i % every_step == 0) and (t > start_time)):
        vxs.append(vx)
        vys.append(vy)
        vzs.append(vz)

    # update coordinates

        x = x + dt * vx
        y = y + dt * vy
        z = z + dt * vz

    #if ((i % every_step == 0) and (t > start_time)):
        xs.append(x)
        ys.append(y)
        zs.append(z)

    print(nSteps)
    filename =str(i)+'neuro.txt'
    thefile = open(filename, 'w')
    for i in range(0, len(xs)-1,1):
        s = str(xs[i])+' '+str(ys[i])+' '+str(zs[i])+' '+str(math.sqrt(xs[i]*xs[i]+ys[i]*ys[i]))+' '+str(dts[i])+'\n'
        thefile.write("%s" %s)
